src/pudl/transform/eia860.py

```diff
@@ -941,3 +941,70 @@ def clean_boiler_emissions_control_equipment_assn_eia860(
     bece_df = bece_df.dropna(subset="emission_control_id_eia")
 
     return bece_df
+
+
+@asset
+def clean_boiler_cooler_assn_eia860(
+    raw_boiler_cooling_eia860: pd.DataFrame,
+) -> pd.DataFrame:
+    """Pull and transform the EIA 860 boiler to cooler ID table.
+
+    Args:
+        raw_boiler_booling_eia860: Raw EIA 860 boiler to cooler ID association table.
+
+    Returns:
+        pd.DataFrame: A cleaned and normalized version of the EIA boiler to cooler ID
+            table.
+    """
+    # Replace empty strings, whitespace, and '.' fields with real NA values
+    bc_assn = pudl.helpers.fix_eia_na(raw_boiler_cooling_eia860)
+    # Replace the report year col with a report date col for the harvesting process
+    bc_assn = pudl.helpers.convert_to_date(
+        df=bc_assn, year_col="report_year", date_col="report_date"
+    )
+    # Drop rows with no cooling ID and just in case, drop duplicate
+    bc_assn = bc_assn.dropna(subset="cooling_id_eia").drop_duplicates()
+
+    return bc_assn
+
+
+@asset
+def clean_boiler_stack_flue_assn_eia860(
+    raw_boiler_stack_flue_eia860: pd.DataFrame,
+) -> pd.DataFrame:
+    """Pull and transform the EIA 860 boiler to stack flue ID table.
+
+    Args:
+        raw_boiler_booling_eia860: Raw EIA 860 boiler to stack flue ID association
+            table.
+
+    Returns:
+        pd.DataFrame: A cleaned and normalized version of the EIA boiler to stack flue
+            ID table.
+    """
+    # Replace empty strings, whitespace, and '.' fields with real NA values
+    bsf_assn = pudl.helpers.fix_eia_na(raw_boiler_stack_flue_eia860)
+    # Replace the report year col with a report date col for the harvesting process
+    bsf_assn = pudl.helpers.convert_to_date(
+        df=bsf_assn, year_col="report_year", date_col="report_date"
+    )
+    # Drop duplicates
+    bsf_assn = bsf_assn.drop_duplicates()
+    # Create a primary key column for stack flue IDs.
+    # There are some stack & flue units that are combined and have just one ID. Others
+    # have unique IDs for each stack & flue unit that may have m:m relationships with
+    # one another. These IDs are part of the primary key for this table, and because we
+    # can't have NA values in the primary key we'll need to create a hybrid column that
+    # turns the seperate stack_id_eia and flue_id_eia values into one stack_flue_id_pudl
+    # value. Anytime there's a stack_flue_id_eia we'll use that.
+    bsf_assn = bsf_assn.assign(
+        stack_flue_id_pudl=lambda x: (
+            x.stack_id_eia.astype("string") + "_" + x.flue_id_eia.astype("string")
+        )
+        .fillna(x.stack_flue_id_eia)
+        .fillna(
+            x.stack_id_eia.astype("string")
+        )  # sometimes there's a stack id and no flue id
+    )
```

Comment:
Do you want to drop records without a PUDL stack/flue ID here, as you do with the other table?

Reply:
That's probably a good idea! Will do.